src/SpecDataHolder.sol

```diff
@@ -26,6 +27,27 @@ contract SpecDataHolder is UUPSUpgradeable, OwnableUpgradeable, ISpecDataHolder
     _disableInitializers();
   }
 
+  // check to see if the _specCreator (msg.sender) owns a badge that's associated with _raftTokenId
+  function doesSenderOwnADaosBadge(address _specCreator, uint256 _raftTokenId) public view returns (bool) {
+    ERC721EnumerableUpgradeable badgesInterface = ERC721EnumerableUpgradeable(badgesAddress);
+    // return badgesInterface.ownerOf(_raftTokenId);
+    // call `balanceOf(_specCreator)` to get the number of badges owned by _specCreator
+    uint256 numBadges = badgesInterface.balanceOf(_specCreator);
+    // for each badge owned by _specCreator, check if it's associated with _raftTokenId
+    for (uint256 i = 0; i < numBadges; i++) {
```

Comment:
Can this be done by the client app and not the contract? And here we just take their badgeTokenID as the argument?

Reply:
Ah yes! That definitely makes sense